File: Simulation/2243.Calculate Digit Sum of a String/Solution.java
class Solution {
    public String digitSum(String s, int k) {
        if (s.length() <= k) {
            return s;
        }
        StringBuilder ans = new StringBuilder();
        int cnt = 0, cur = 0;
        for (char c : s.toCharArray()) {
            cur += c - '0';
            if (++cnt == k) {
                ans.append(String.valueOf(cur));
                cnt = 0;
                cur = 0;
            }
        }
        if (cnt > 0) {
            ans.append(String.valueOf(cur));
        }
        return digitSum(ans.toString(), k);
    }
}

File: Simulation/2243.Calculate Digit Sum of a String/Solution.py
class Solution:
    def digitSum(self, s: str, k: int) -> str:
        if len(s) <= k:
            return s
        ans = []
        cnt = cur = 0
        for c in s:
            cur += ord(c) - ord('0')
            cnt += 1
            if cnt == k:
                ans.append(str(cur))
                cnt = cur = 0
        if cnt:
            ans.append(str(cur))
        return self.digitSum("".join(ans), k)